[INTEL MKL] Integrated MKL to TF conversion op with MKL-DNN v1.x

bhavani-subramanian opened the pull request.

bhavani-subramanian committed:
Integrated MKL to TF conversion op with MKL-DNN v1.x

penpornk commented on tensorflow/core/kernels/mkl_tfconv_op.h:
```diff
@@ -101,13 +110,34 @@ class MklToTfOp : public OpKernel {
+        // Insert reorder between MKL layout and TensorFlow layout.
+        // TODO(bhavanis): Refactor CheckReorderToOpMem() to directly insert
+        // the reorder primitive
+        bool status = input.CheckReorderToOpMem(output_tf_md, output_tensor,
```
The statement can be inside `OP_REQUIRES`.  Same for the one below.

bhavani-subramanian replied:
Done

penpornk reviewed (changes requested):
Thank you for the PR!

bhavani-subramanian committed:
1. Used the correct version of CheckReorderToOpMem() that both creates and executes reorder primitive.
2. Defined NET_ARGS_PTR so that ExecutePrimitive() calls can be common to both 0.x and 1.x versions.
3. Directly executing conditions within OP_REQUIRES.

bhavani-subramanian commented:
@penpornk I have addressed your review comments. Please take a look. Thank you!

penpornk commented on tensorflow/core/kernels/mkl_tfconv_op.h:
```diff
@@ -37,13 +37,17 @@ limitations under the License.
+
 #ifdef ENABLE_MKLDNN_V1
 #define ENGINE_CPU engine::kind::cpu
+#define NET_ARGS_PTR &net_args
```
`NET_ARGS_PTR` is not used in this file.

bhavani-subramanian replied:
Yes, you're right. I removed this macro.

penpornk commented on tensorflow/core/kernels/mkl_tfconv_op.h:
```diff
@@ -110,32 +114,16 @@ class MklToTfOp : public OpKernel {
-            context, status,
-            errors::Internal("MklToTfOp: Failed to create input reorder"));
-
-        ExecutePrimitive(net, &net_args, cpu_engine);
```
`ExecutePrimitive` is not called anymore. Is this intentional?

bhavani-subramanian replied:
Actually, this version of `CheckReorderToOpMem()` both creates and executes the reorder primitive: i.e. https://github.com/tensorflow/tensorflow/blob/063a8303f69645e1da610ef8a5bbfcfc782e42bb/tensorflow/core/util/mkl_util.h#L1680 calls https://github.com/tensorflow/tensorflow/blob/063a8303f69645e1da610ef8a5bbfcfc782e42bb/tensorflow/core/util/mkl_util.h#L1613 which in turn executes reorder.

penpornk replied:
Got it. Thank you for the explanation! :)

penpornk reviewed (commented):
Thank you for the edits! I have a few questions please.

bhavani-subramanian committed:
Removed unused macro.
